internal/mage/engine.go

```diff
@@ -48,3 +52,165 @@ func (t Engine) Lint(ctx context.Context) error {
 		}).ExitCode(ctx)
 	return err
 }
+
+const (
+	sdkHelper      = "dagger-sdk-helper"
+	buildkitRepo   = "github.com/moby/buildkit"
+	buildkitBranch = "v0.10.5"
+	// TODO: placeholder until real one exists
+	engineImageRef = "eriksipsma/dagger-test:bootstrap"
+)
+
+func engineContainer(c *dagger.Client, arches, oses []string) []*dagger.Container {
+	buildkitRepo := c.Git(buildkitRepo).Branch(buildkitBranch).Tree()
+
+	var platformVariants []*dagger.Container
+	for _, arch := range arches {
+		buildkitBase := c.Container(dagger.ContainerOpts{
+			Platform: dagger.Platform("linux/" + arch),
+		}).Build(buildkitRepo)
+		for _, os := range oses {
+			// include each helper for each arch too in case there is a
+			// client/server mismatch
+			for _, arch := range arches {
+				helperBin := util.GoBase(c).
+					WithEnvVariable("GOOS", os).
+					WithEnvVariable("GOARCH", arch).
+					Exec(dagger.ContainerExecOpts{
+						Args: []string{"go", "build", "-o", "./bin/" + sdkHelper, "-ldflags", "-s -w", "/app/cmd/sdk-helper"},
+					}).
+					File("./bin/" + sdkHelper)
+				buildkitBase = buildkitBase.WithFS(
+					buildkitBase.FS().WithFile("/usr/bin/"+sdkHelper+"-"+os+"-"+arch, helperBin),
+				)
+			}
+		}
+		platformVariants = append(platformVariants, buildkitBase)
+	}
+
+	return platformVariants
+}
+
+func (t Engine) Release(ctx context.Context) error {
```

Comment:
nit: I was thinking of standardizing this to `sdk:*:publish` -- to avoid confusion with releasing (cutting the tag etc)